packages/share_plus/share_plus/android/src/main/kotlin/dev/fluttercommunity/plus/share/MethodCallHandler.kt

```diff
@@ -8,72 +8,68 @@ import java.io.IOException
-                }
-            }
-            "share", "shareWithResult" -> {
-                expectMapArguments(call)
-                if (isWithResult && !manager.setCallback(result)) return
+        if (isWithResult && !manager.setCallback(result)) return
 
-                // Android does not support showing the share sheet at a particular point on screen.
-                share.share(
-                    call.argument<Any>("text") as String,
-                    call.argument<Any>("subject") as String?,
-                    isWithResult,
-                )
+        try {
+            when (call.method) {
+                "shareUri" -> {
+                    share.share(
+                        call.argument<Any>("uri") as String, subject = null, withResult = false
+                    )
+                    success(isWithResult, isResultRequested, result)
+                }
 
-                if (!isWithResult) {
-                    if (isResultRequested) {
-                        result.success("dev.fluttercommunity.plus/share/unavailable")
-                    } else {
-                        result.success(null)
-                    }
+                "share", "shareWithResult" -> {
+                    share.share(
+                        call.argument<Any>("text") as String,
+                        call.argument<Any>("subject") as String?,
+                        isWithResult,
+                    )
+                    success(isWithResult, isResultRequested, result)
                 }
-            }
-            "shareFiles", "shareFilesWithResult" -> {
-                expectMapArguments(call)
-                if (isWithResult && !manager.setCallback(result)) return
 
-                // Android does not support showing the share sheet at a particular point on screen.
-                try {
+                "shareFiles", "shareFilesWithResult" -> {
                     share.shareFiles(
                         call.argument<List<String>>("paths")!!,
                         call.argument<List<String>?>("mimeTypes"),
                         call.argument<String?>("text"),
                         call.argument<String?>("subject"),
                         isWithResult,
                     )
-
-                    if (!isWithResult) {
-                        if (isResultRequested) {
-                            result.success("dev.fluttercommunity.plus/share/unavailable")
-                        } else {
-                            result.success(null)
-                        }
-                    }
-                } catch (e: IOException) {
-                    result.error("Share failed", e.message, null)
+                    success(isWithResult, isResultRequested, result)
                 }
+
+                else -> result.notImplemented()
+            }
+        } catch (e: Throwable) {
+            manager.clear()
+            result.error("Share failed", e.message, e)
+        }
```

Comment:
Catch _all_ errors from share methods, pass them up to the Flutter layer, but before also call to `manager.clear()` to avoid the deadlock state in `ShareSuccessManager`.

This can be tested simply by calling

```dart
    await Share.shareXFiles(
      [XFile('/wrong/file')],
      text: text,
    );
```
then the user sees something like
```
E/flutter (29015): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: PlatformException(Share failed, /wrong/file: The source file doesn't exist.,
```
however, they will be able to use the share methods again later on because `manager.clear()` was called.